### PR TITLE
[RNBW-2464] Favorites Fixes

### PR DESCRIPTION
**Pull request by derHowie**
Fixes RNBW-2464

## What changed (plus any additional context for devs)
There were several things causing favorites to behave incorrectly on DiscoverSearch.

1. `favorite` bool was missing from favorites meta data completely
2. ETH values were improperly keyed as WETH in some cases (weirdness Pedro saw)
3. Not related to originally reported issue, but noticed that removing all favorites causes permanent skeleton on DiscoverSearch. This is a production bug I was able to reproduce in App Store. I put in a fix for this.
4. Fixed a very specific case where you could extract duplicate key props for coin rows when adding a token from the curated list to the favorites list. This only occurred when the curated token's index was the same as the index of its new position in the favorites list. (e.g. click on 2nd curated token with 1 favorite in your list, which then creates duplicate keyed row in 2nd position of favorites)

## PoW (screenshots / screen recordings)
Wrapped Bitcoin

## Dev checklist for QA: what to test

- adding / removing favorites and then verify correct list displayed in discover search and swap modal
- test fresh above at least once with fresh user
- test above at least once with existing user

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Comment by skylarbarrera** [comment]
everything looks fixed but im seeing one smol regression, we use the `currencySelectionList` on the swap flow too so we need to make sure we handle both scenarios: https://cloud.skylarbarrera.com/Screen-Shot-2022-02-09-11-58-13.53.png

**Review by skylarbarrera — APPROVED** [review]
👌

**Comment by peduarte** [comment]
It's working much better for me ✨ that's good stuff Chris. 

While testing these changes, I noticed that when I unfavourited an item, and then left, came back to the screen, it was still there. See video below:

https://user-images.githubusercontent.com/372831/154113565-a32112bf-26e5-4dd3-9b9d-4bdd46b78e09.mp4

I managed to replicate this a few times, but sometimes it worked. It also seems to happen mainly with Unisocks. The only thing that comes to mind is that maybe the UI does some sort of optimistic update, but the backend isn't yet updated by the time the component unmounts. I'm sorry I dont have much more to add, but hope this helps. Happy to test anything else if you need.

**Comment by derHowie** [comment]
Nice catch, @peduarte. I was able to reproduce the issue you documented on the App Store version and made a ticket to fix here: [RNBW-2567](https://linear.app/rainbow/issue/RNBW-2567/addremove-favorites-inconsistencies-on-currencyselectmodal).

File: src/hooks/useUniswapCurrencyList.ts [hunk 1]
```diff
@@ -185,27 +185,30 @@ const useUniswapCurrencyList = (searchQuery: string) => {
...
         });
-        if (curatedAssets.length) {
-          list.push({
-            data: curatedAssets,
-            title: tokenSectionTypes.verifiedTokenSection,
-            useGradientText: IS_TESTING === 'true' ? false : true,
-          });
-        }
+      }
+      const curatedAssets = getCurated();
```
Comment (jinchung) [inline src/hooks/useUniswapCurrencyList.ts:192]: nit not a blocker: is there any reason `getCurated` needs to be a callback instead of memo?

**Review by jinchung — APPROVED** [review]
🌮

**Review by peduarte — APPROVED** [review]
Apart from the issue in my earlier comment, it looks good 🚀

**Review by ibrahimtaveras00 — APPROVED** [review]
All looks good 👍🏽 

I saw an issue on android but it currently happens on playstore, so I created [RNBW-2575](https://linear.app/rainbow/issue/RNBW-2575/cant-addremove-favorite-on-swap-list-on-android)

**Pull request merged**